File: 34. Find First and Last Position of Element in Sorted Array/find_first_and_last_position_of_element_in_sorted_array.kt
fun first_and_last (element: Int, list: Array<Int>) : Array<Int>{
	var find_first: Boolean = true;
	val len_list: Int = list.size;
	var out: Array<Int> = arrayOf(-1, -1);
	list.sort();
	
	for (i in 0 until len_list){
		if (element == list[i]){
			if (find_first){
				out[0] = i;
				out[1] = i;
				find_first = false;
			} else {
				out[1] = i;
			}
		}
	}
	
	return (out);
}

fun main () {
	var nums: Array<Int> = arrayOf(5,7,7,8,8,10,2);
	var target: Int = 8;
	
	var positions: Array<Int> = first_and_last (target, nums);
	
	print("This is the handy work of Java\n\n\n[");
	print(positions[0]);
	print(", ");
	print(positions[1]);
	println("]\n\n");
}

File: 34. Find First and Last Position of Element in Sorted Array/find_first_and_last_position_of_element_in_sorted_array.rs
fn first_and_last (element: isize, list: &mut [isize], out: &mut [isize]) -> (){
	let mut find_first: bool = true;
	let len_list: usize = list.len();
	
	list.sort();
	
	for i in 0..len_list{
		if (element == list[i]){
			if (find_first){
				out[0] = i as isize;
				out[1] = i as isize;
				find_first = false;
			} else {
				out[1] = i as isize;
			}
		}
	}
}

fn main () {
	let mut nums = [5,7,7,8,8,10];
	let target: isize = 8;
	let mut positions: [isize; 2] = [-1, -1];
	
	first_and_last (target, &mut nums, &mut positions);
	
	println!("\n\nThis is the handy work of Rust\n\n");
	println!("{:?}\n\n\n", positions);
}
